framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2653,6 +2653,58 @@ public AnnotatedWildcardType getUninferredWildcardType(AnnotatedTypeVariable typ
         return wctype;
     }
 
+    /**
+     * Returns the most specific base type of {@code annotatedTypeMirror} whose erasure type
+     * is the upper bound of {@code wildcard}, if {@code wildcard}'s upper
+     * bound is a super type of {@code annotatedTypeMirror}.
+     *
+     * Otherwise, returns {@code annotatedTypeMirror} unmodified.
+     *
+     * For example:
+     * <pre>
+     * wildcard := @NonNull ? extends @NonNull Object
+     * annotatedTypeMirror := @Nullable String
+     *
+     * widenToUpperBound(annotatedTypeMirror, wildcard) returns @Nullable String
+     * </pre>
+     * The widened type should only be used for typing checks that require it. Using the widened type
```

Comment:
Yes it is always a super type of annotatedTypeMirror 
Yes, they stay unchanged.  I updated the comment